internal/firmware/info_store.go

```diff
@@ -0,0 +1,31 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package firmware
+
+// StoreInfo holds firmware metadata for which the store is the
+// canonical source.
+//
+// It can be marshalled and will be stored in the system state for
+// each installed firmware so it needs to be evolved carefully.
+type StoreInfo struct {
```

Comment:
Any reason this needs to go into its own file? It seems to me this would be clearer right alongside `type Info`.

Reply:
Fixed.